.github/workflows/static-checks.yaml

```diff
@@ -0,0 +1,126 @@
+on:
+  workflow_call:
+    inputs:
+      runs-on:
+        description: 'A tag to indicate which runner to use'
+        required: true
+        type: string
+      gochannel:
+        description: 'The snap store channel to use to install the go snap'
+        required: true
+        type: string
+
+jobs:
+  static-checks:
+    runs-on: ${{ inputs.runs-on }}
+    env:
+      GOPATH: ${{ github.workspace }}
+      # Set PATH to ignore the load of magic binaries from /usr/local/bin And
+      # to use the go snap automatically. Note that we install go from the
+      # snap in a step below. Without this we get the GitHub-controlled latest
+      # version of go.
+      PATH: /snap/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:${{ github.workspace }}/bin
+      GOROOT: ""
+      GITHUB_PULL_REQUEST_TITLE: ${{ github.event.pull_request.title }}
+      BASE_REF: ${{ github.base_ref }}
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+        # needed for git commit history
+        fetch-depth: 0
+        # NOTE: checkout the code in a fixed location, even for forks, as this
+        # is relevant for go's import system.
+        path: ./src/github.com/snapcore/snapd
+
+    # Fetch base ref, needed for golangci-lint
+    - name: Fetching base ref ${{ github.base_ref }}
+      run: |
+        cd ${{ github.workspace }}/src/github.com/snapcore/snapd
+        git fetch origin ${{ github.base_ref }}:${{ github.base_ref }}
+
+    - name: Download and install Debian dependencies
+      # Github does not allow variables in "uses"; this has to be a hard-coded path
+      uses: ./src/github.com/snapcore/snapd/.github/workflows/actions/download-install-debian-deps
+      with:
+        snapd-src-dir: "${{ github.workspace }}/src/github.com/snapcore/snapd"
+
+    # golang latest ensures things work on the edge
+    - name: Install the go snap
+      run: |
+          sudo snap install --classic --channel=${{ inputs.gochannel }} go
```

Comment:
I wrote https://github.com/canonical/snapd-wsl-tests/tree/main/.github/actions/download-snap and https://github.com/canonical/snapd-wsl-tests/tree/main/.github/actions/install-offline-snap that we might want to use later on. One advantage is that they use GitHub cache so installation is very very fast, even if the store is slow or down completely. They'd have to be ported a little from powershell but the idea could stand.

Reply:
Nice! I'll look into adding that in the cleanup task. That sounds great.